### PR TITLE
rate and write review

### PR DESCRIPTION
**Pull request by MuhammadTamzid**

### CODE REVIEW

File: app/models/review.rb [hunk 1]
```diff
@@ -2,4 +2,11 @@ class Review < ActiveRecord::Base
   belongs_to :user
   belongs_to :book
   has_many :comments
+
+  validates :book_id,
```
Comment (dieunb) [inline app/models/review.rb]: tell me, difference between `validates :book_id` and `validates :book`. Which way should you use?

Comment (MuhammadTamzid) [inline app/models/review.rb]: Yes should use validates :book

**Comment by sujoydatta** [comment]
ok

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    if @review.save
+      respond_to do |format|
+        format.html do
+          flash[:success] = t "add_review_sucessful_message"
+          redirect_to @review.book
+        end
+        format.js
+      end
+    else
+      flash[:success] = t "add_review_fail_message"
```
Comment (dieunb) [inline app/controllers/reviews_controller.rb]: should use ajax to show errors

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      redirect_to @review.book
```
Comment (dieunb) [inline app/controllers/reviews_controller.rb]: should use ajax to show errors

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  def create
+    @review.user = current_user
+    if @review.save
+      respond_to do |format|
+        format.html do
+          flash[:success] = t "add_review_sucessful_message"
+          redirect_to @review.book
+        end
+        format.js
+        format.json {render json: {message: t(:add_review_sucessful_message)}}
```
Comment (dieunb) [inline app/controllers/reviews_controller.rb]: no need

File: app/controllers/reviews_controller.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      end
+    else
+      respond_to do |format|
+        format.html do
+          flash[:danger] = t "add_review_fail_message"
+          redirect_to @review.book
+        end
+        format.js
+        format.json {render json: {message: t(:add_review_fail_message)}}
```
Comment (dieunb) [inline app/controllers/reviews_controller.rb]: no need for rails ajax

File: app/views/reviews/create.js.erb [hunk 6]
```diff
@@ -0,0 +1 @@
+$("#review_form").hide();
```
Comment (dieunb) [inline app/views/reviews/create.js.erb]: missing error case

**Pull request merged**